tests/test_event_detection.py

```diff
@@ -1,7 +1,180 @@
 from scipy.io import loadmat
 
+sys.path.append('/gpfs/home/ddiesbur/SpectralEvents')
+import spectralevents as se
+
+def test_event_comparison(method):
+    '''Test whether the output from MATLAB and Python event detection scripts align
+
+    Parameters
+    ----------
+    method : float
+    The event detection method (1, 2, or 3) used to detect spectral events.
+
+    Notes
+    -----
+    Currently only supports testing method 1 and 3, with method 2 to be added
+    to the Python event detection functions in the future. Python output is
+    tested against the output of MATLAB scripts from Shin et al., 2017. Detection
+    of the same number of beta events within each trial is required, and tolerable
+    margins of deviation for the other parameters.
+    '''
+
+    # Define tolerable deviance for spectral event characteristics
+    dev_max_time = 5  # +-5ms
+    dev_max_freq = 1  # +-1Hz
+    dev_low_freq = 1  # +-1Hz
+    dev_high_freq = 1  # +-1Hz
+    dev_onset = 10  # +-10ms
+    dev_offset = 10  # +-10ms
+    dev_fom_pow = 0.5  # +-0.5 FOM
+
+    # Load MATLAB data
+    data_dir = os.getcwd()
+    if method == 1:
+        matlab_file = loadmat(
+            op.join(data_dir, 'tests', 'beta_events_shin_2017.mat'))
+    #elif method == 3:
+    #    matlab_file = loadmat(
+    #        op.join(data_dir, 'tests', 'beta_events_shin_2017_method3.mat'))
+    else:
+        raise ValueError('Unsupported method! Please use 1') #or 3')
+    mat_all_evs = matlab_file['event_times'][:, 0]
+
+    # Get python events by running find_events on demo data
+    # <----------SIMPLIFY BY GETTING PRE-SAVED PICKLE FILES FOR THIS INSTEAD----------->
+    subj_ids = [str(id) for id in range(1, 10 + 1)]  # subject IDs 1-10
+
+    # set parameters
+    samp_freq = 600
+    n_times = 600
+    # fequency values (Hz) over which to calculate TFR
+    freqs = list(range(1, 60 + 1))
+    times = np.arange(n_times) / samp_freq  # seconds
+    event_band = [15, 29]  # beta band (Hz)
+    thresh_fom = 6.0  # factor-of-the-median threshold
+
+    # get python evs
+    py_all_evs = list()
+    for _, id_val in enumerate(subj_ids):
+        fname = op.join(
+            data_dir, 'data','prestim_humandetection_600hzMEG_subject' + id_val + '.mat')
+        raw_data = loadmat(fname)
+        data = raw_data['prestim_raw_yes_no']  # MEG time series (trials x samples)
+        # calculate TFR
+        tfrs = se.tfr(data, freqs, samp_freq)
+
+        # find events
+        subj_evs = se.find_events(tfr=tfrs, times=times, freqs=freqs, event_band=event_band,
+                                  thresholds=None, threshold_FOM=thresh_fom, find_method=method)
+        py_all_evs.append(subj_evs)
+
+    # put from cell array/dict into sets
+    py_evs = np.array([])
+    for subj_idx, id_val in enumerate(subj_ids):
+        py_evs = np.append(py_evs,np.array([len(trial) for trial in py_all_evs[subj_idx]]))
+    mat_evs = np.array([trial.shape[1] for trial in mat_all_evs])
```

Comment:
For better code readability, I'd recommend only using `list` objects instead of `ndarray` until you absolutely need more complex functionality (i.e., loop over trials while appending to the `py_evs` list, then convert to it to an array).